engine/src/main/java/org/terasology/world/block/IBlockRegion.java

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+package org.terasology.world.block;
+
+import org.joml.AABBf;
+import org.joml.LineSegmentf;
+import org.joml.Matrix4fc;
+import org.joml.Planef;
+import org.joml.Rayf;
+import org.joml.Spheref;
+import org.joml.Vector2f;
+import org.joml.Vector3f;
+import org.joml.Vector3fc;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+import org.terasology.entitySystem.entity.EntityRef;
+
+import java.util.Iterator;
+import java.util.Optional;
+
+public interface IBlockRegion {
+    public static BlockRegion invalid();
```

Comment:
Something like this required to build the union over multiple "things" (positions, regions, ...) into a new region (initial value).

Reply:
No reason to have a method for invalid.  Either make the constructor methods public or define as a constant in blockregion.

Reply:
Making `new BlockRegion()` would undermine the design principle to ensure that regions are valid.

Having a static `BlockRegion.INVALID` without an immutable interface would allow to change `BlockRegion.INVALID` itself - which means it would not be a constant...

That's why I went for a static method here.

Reply:
C in blockregionc is constant. So invalid is guarded by the interface.